processor/src/main/java/com/umairjavid/kombind/processor/AdapterProcessor.kt

```diff
@@ -0,0 +1,118 @@
+package com.umairjavid.kombind.processor
+
+import com.google.auto.service.AutoService
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asTypeName
+import com.umairjavid.kombind.anontation.SimpleKombindAdapter
+import java.io.File
+import javax.annotation.processing.AbstractProcessor
+import javax.annotation.processing.Messager
+import javax.annotation.processing.ProcessingEnvironment
+import javax.annotation.processing.Processor
+import javax.annotation.processing.RoundEnvironment
+import javax.lang.model.SourceVersion
+import javax.lang.model.element.TypeElement
+import javax.lang.model.element.VariableElement
+import javax.tools.Diagnostic
+import kotlin.reflect.KClass
+
+@AutoService(Processor::class)
+class AdapterProcessor : AbstractProcessor() {
+    lateinit var messenge: Messager
+
+    override fun init(env: ProcessingEnvironment?) {
+        super.init(env)
+        messenge = env!!.messager
+    }
+    override fun getSupportedAnnotationTypes() = mutableSetOf(SimpleKombindAdapter::class.java.name)
+
+    override fun getSupportedSourceVersion() = SourceVersion.latestSupported()
+
+    override fun process(typeElemnts: MutableSet<out TypeElement>?, roundEnv: RoundEnvironment?): Boolean {
+        val env = roundEnv!!.getElementsAnnotatedWith(SimpleKombindAdapter::class.java)
+        env.forEach {
+            val layoutRes = it.getAnnotation(SimpleKombindAdapter::class.java).layoutRes
+            if (it is VariableElement)
+                generateAdapterClass(it, layoutRes)
+        }
+        return false
+    }
+
+    fun generateAdapterClass(element: VariableElement, layoutRes: Int) {
+        if (!element.asType().asTypeName().isMutableLiveArraylist()) {
+            messenge.printMessage(Diagnostic.Kind.ERROR, "Needs to be of type MutableArrayList")
+            return
+        }
+
+        val kombindPackage = "com.umairjavid.kombind.ui"
+        val appPackageName = processingEnv.elementUtils.getPackageOf(element).simpleName.toString()
+        val kombindAdapterName = "KombindAdapter"
+        val viewHolder = "KombindAdapter.ViewHolder"
+        val fileName = element.simpleName
+        val className = "Kombind_${fileName}_Adapter"
```

Comment:
in that case, can you change this? first it seems like it's not really `fileName` but `elementName`
second, could you do: `"Kombind${elementName.capitalize()}Adapter"`

Reply:
also how does this handle duplicate classes with the same name?

Reply:
Thats a good question lol. Perhaps I'll save the names in a set, and if the name has already been used I will add the enclosing class  name as a namespace. Or have the user specify  the name, with the annotation and error if the name has been used. What do you think?

Reply:
@ujavid Ok just updated this .  Now the generated class will including the enclosing class name of the property. 
for instance : In MainState.k there is @SimpleKombindAdapter val items = MutableLiveArrayList
This generated a class called, KombindItemsAdapter_MainState.
It shouldn't be possible to have duplicate names with that signature structure,